### DISCUSSION TITLE
Modify non-linear variable at every timestep

### DISCUSSION THREAD

**Discussion by rayaprolu143** [idaholab/moose#27291] [Q&A Modules: Phase field] [answered]
Hello MOOSE users,

I want to modify a nonlinear variable at every timestep, such as the chemical potential of a vacancy inside the void. So far, I have tried using the multi apps for this end. Although I successfully achieved the objective, there was a 3 to 4 order reduction in the time step to achieve the convergence. I would like to know if there is any other better way to modify the nonlinear variable at every timestep.

Thank you,
Sreekar

**Comment by GiudGiud** [comment]
Hello

are you performing fixed point iterations between the main app and the children applications?

if you only perform a single step of coupling, the time integration scheme is effectively explicit in one (or more) variable, hence it can require smaller time steps 

Guillaume

**Reply by GiudGiud** [reply, marked as answer]
And otherwise you can modify non linear variables inside user objects. They are not too difficult to create for that purpose.

but I would expect exactly the same results as a multiapp
